client/rest-high-level/src/main/java/org/elasticsearch/client/security/InvalidateApiKeyRequest.java

```diff
@@ -0,0 +1,152 @@
+import org.elasticsearch.client.ValidationException;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * Request for invalidating API key(s) so that it can no longer be used
+ */
+public final class InvalidateApiKeyRequest implements Validatable, ToXContentObject {
+
+    private final String realmName;
+    private final String userName;
+    private final String id;
+    private final String name;
+
+    // pkg scope for testing
+    InvalidateApiKeyRequest(@Nullable String realmName, @Nullable String userName, @Nullable String apiKeyId,
+            @Nullable String apiKeyName) {
+        this.realmName = realmName;
+        this.userName = userName;
+        this.id = apiKeyId;
+        this.name = apiKeyName;
+    }
+
+    public String getRealmName() {
+        return realmName;
+    }
+
+    public String getUserName() {
+        return userName;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Creates invalidate API key request for given realm name
+     * @param realmName realm name
+     * @return {@link InvalidateApiKeyRequest}
+     */
+    public static InvalidateApiKeyRequest usingRealmName(String realmName) {
+        return new InvalidateApiKeyRequest(realmName, null, null, null);
+    }
+
+    /**
+     * Creates invalidate API key request for given user name
+     * @param userName user name
+     * @return {@link InvalidateApiKeyRequest}
+     */
+    public static InvalidateApiKeyRequest usingUserName(String userName) {
+        return new InvalidateApiKeyRequest(null, userName, null, null);
+    }
+
+    /**
+     * Creates invalidate API key request for given realm and user name
+     * @param realmName realm name
+     * @param userName user name
+     * @return {@link InvalidateApiKeyRequest}
+     */
+    public static InvalidateApiKeyRequest usingRealmAndUserName(String realmName, String userName) {
+        return new InvalidateApiKeyRequest(realmName, userName, null, null);
+    }
+
+    /**
+     * Creates invalidate API key request for given api key id
+     * @param apiKeyId api key id
+     * @return {@link InvalidateApiKeyRequest}
+     */
+    public static InvalidateApiKeyRequest usingApiKeyId(String apiKeyId) {
+        return new InvalidateApiKeyRequest(null, null, apiKeyId, null);
+    }
+
+    /**
+     * Creates invalidate API key request for given api key name
+     * @param apiKeyName api key name
+     * @return {@link InvalidateApiKeyRequest}
+     */
+    public static InvalidateApiKeyRequest usingApiKeyName(String apiKeyName) {
+        return new InvalidateApiKeyRequest(null, null, null, apiKeyName);
+    }
+
+    @Override
+    public Optional<ValidationException> validate() {
```

Comment:
In the HLRC effort over the last few months, I think we made a conscious effort to not override `validate()` but instead do the validation in the constructor.